modules/openxr/scene/openxr_composition_layer.cpp

```diff
@@ -212,7 +262,19 @@ void OpenXRCompositionLayer::_reset_fallback_material() {
 		return;
 	}
 
-	if (layer_viewport) {
+	if (enable_hole_punch && !Engine::get_singleton()->is_editor_hint() && is_natively_supported()) {
+		Ref<ShaderMaterial> material = fallback->get_surface_override_material(0);
```

Comment:
I might be reading this wrong but what happens if I need the fallback and I have hole punch enabled? Wouldn't this override the material that renders my viewport?

Reply:
No, if you were on a platform that didn't natively support composition layers and you needed the "real" fallback mesh, then this code block wouldn't run because `is_natively_supported()` would return `false`, and you'd end up in the next code block which sets up the `ViewportTexture`